core/src/main/scala/za/co/absa/spline/core/SparkLineageInitializer.scala

```diff
@@ -33,12 +38,18 @@ import scala.util.control.NonFatal
   */
 object SparkLineageInitializer extends Logging {
 
-  def enableLineageTracking(sparkSession: SparkSession): SparkSession = {
+  def enableLineageTracking(sparkSession: SparkSession): SparkSession =
     SparkSessionWrapper(sparkSession).enableLineageTracking()
-  }
 
-  def enableLineageTracking(sparkSession: SparkSession, configurer: SplineConfigurer): SparkSession = {
+  def enableLineageTracking(sparkSession: SparkSession, configurer: SplineConfigurer): SparkSession =
     SparkSessionWrapper(sparkSession).enableLineageTracking(configurer)
+
+  def createSplineOrNoopQueryExecutionListener(sparkSession: SparkSession): QueryExecutionListener =
+    SparkSessionWrapper(sparkSession).createSplineOrNoopQueryExecutionListener()
+
+  class NoOpQueryExecutionListener extends QueryExecutionListener {
```

Comment:
Why do you need this class? Wouldn't a simple `Option` suffice?

Reply:
I need to delegate to actual implementation. It would be possible to have an ```Option#isDefined``` check on each call to CodelessQueryExecutionListener.

Reply:
What's wrong with `Option.forEach`?